server/routes/admin/articles/articles.js

```diff
@@ -24,28 +24,46 @@ router.get('/add', (req, res, next) => {
     }
     categoryClient.findCategories(callback);
 });
-
+//////////////////////////////////////////////
+/////////////////////////////////////////////
 router.post('/add', function (req, res, next) {
-    const query = req.body;
-    if (req.files) {
-        let articleImage = req.files.articleImage,
-            filename = articleImage.name;
-        articleImage.mv('../server/public/images/' + filename, function (err) {
-            if (err) {
-                console.log(err)
-                res.send(' error accurd');
-            } else {
-                console.log('done');
-            }
-        });
-    }
-
-    const callBack = (data) => {
-        res.redirect('/')
-    }
-    articleClient.addArticle(query, callBack)
-});
+    let query = {}
+    if (Object.keys(req.files).length > 0 && (req.files).constructor === Object) 
```

Comment:
you don't need `(req.files).constructor === Object`

Reply:
this whole `if` logic can be simplified. Just have a query with all the fields except articleImage .. and add articleImage if it's there